.github/workflows/triviy.yml

```diff
@@ -0,0 +1,84 @@
+name: Trivy scan
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '46 8 * * 0'
+  pull_request:
+    branches: ["master", "release/**"]
+
+# Declare default permissions as nothing.
+permissions: {}
+
+jobs:
+  trivy-scan:
+    name: Trivy scan
+    runs-on: ubuntu-22.04
+
+    steps:
+      - name: Install trivy package
+        run: |
+          sudo apt-get install wget apt-transport-https gnupg lsb-release
+          wget -qO - https://aquasecurity.github.io/trivy-repo/deb/public.key | sudo apt-key add -
+          echo deb https://aquasecurity.github.io/trivy-repo/deb $(lsb_release -sc) main | \
+          sudo tee -a /etc/apt/sources.list.d/trivy.list
+          sudo apt-get update
+          sudo apt-get install trivy
+
+      - name: Checkout code
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11  # v4.1
+        with:
+          persist-credentials: false
+
+      - name: Checkout latest trivy configuration
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11  # v4.1.1
+        with:
+          ref: grom72/trivy  # 'grom72/trivy' to be changed to 'master' before the merge
+          path: trivy
+          persist-credentials: false
+
+      - name: Update trivy configuration
+        run: |
+          cp -f -r ./trivy/utils/trivy ./utils
+          rm -rf ./trivy
+
+      - name: Scan with trivy
+        run: |
+          trivy fs -c utils/trivy/trivy.yaml -f table --dependency-tree \
+          --skip-files "src/client/java/hadoop-daos/pom.xml" \
+          --show-suppressed --exit-code 1 .
+
+      # generate trivy report only if no errors detected
+      - name: Generate trivy report file extension
+        # github.base_ref-based for PR and github.ref_name-based for schedule/workflow_dispatch
```

Comment:
Does using landing-time testing instead of scheduled resolve this either/or for `github.base_ref`/`github.ref_name`?

In either case, see below for a more concise solution.

Reply:
I do not want to block PR because of detected vulnerabilities, as the solution is to update the corrupted package to the proper version or, like what we did with Hadoop, create a suppression. 

When we have this PR landing it will be great to add a status badge to monitor Trivy scan as we have it done for other regular actions:

[![License](https://img.shields.io/badge/License-BSD--2--Clause--Patent-blue.svg)](./LICENSE)
[![Coverity Scan Build Status](https://img.shields.io/coverity/scan/3015.svg)](https://scan.coverity.com/projects/daos-stack-daos)
[![Build](https://github.com/daos-stack/daos/actions/workflows/ci2.yml/badge.svg)](https://github.com/daos-stack/daos/actions/workflows/ci2.yml)
[![Codespell](https://github.com/daos-stack/daos/actions/workflows/spelling.yml/badge.svg)](https://github.com/daos-stack/daos/actions/workflows/spelling.yml)
[![Doxygen](https://github.com/daos-stack/daos/actions/workflows/doxygen.yml/badge.svg)](https://github.com/daos-stack/daos/actions/workflows/doxygen.yml)

Reply:
The problem is that the Trivy scan might fail not because of new code but because of a new vulnerability detected in some unrelated to the PR code.
There are two goals (from the SDLe process perspective):
- prevent any new vulnerability from being introduced by developers (rare case)
- detect new vulnerabilities that are detected and reported by subcomponent owners

The second

Reply:
> The problem is that the Trivy scan might fail not because of new code but because of a new vulnerability detected in some unrelated to the PR code.

Yes, that is understood.  As described above, we already accept this process model with our other checks and validations.

The problem is that we do not have processes that support allowing new (even unrelated) issues to "slide" and land so that they are addressed _later_.  Nobody ever owns fixing those and they continue to pile up until the pile is so big that it's a nightmare.  So the only process that works for us is to have new -- even unrelated issues -- block progress, forcing somebody to own getting it fixed.